app/src/vuex/modules/account.js

```diff
@@ -42,6 +44,23 @@ export default ({ apollo }) => {
         }
       })
     },
+    async listenToIdTokenChanges({ commit }) {
+      const Auth = (await getFirebase()).auth()
+      await Auth.onIdTokenChanged(async (user) => {
+        if (user) {
+          // user is already signed in since we are handling that with onAuthStateChanged
+          commit(`setUserInformation`, user)
+
+          // retrieving refreshed idToken
+          const idToken = await user.getIdToken(/* forceRefresh */ true)
+
+          // really important part, store idToken in localstorage so Apollo gets the newest
+          localStorage.setItem(`auth_token`, idToken)
+          // make sure new authorization token get added to header
+          apollo.cache.reset()
+        }
```

Comment:
is there a case where the user is null and we need to remove the auth token?

Reply:
the problem is not if the user is `null`, but that apollo has already cached the old idToken when we do
```
    // Retrieve the authorization token from local storage.
    const token = localStorage.getItem("auth_token")
```
in `createApolloClient` in `apollo.js`